1-sprint-mission/src/main/java/com/sprint/mission/discodeit/controller/MessageController.java

```diff
@@ -1,90 +1,79 @@
-import io.swagger.v3.oas.annotations.tags.Tag;
+import com.sprint.mission.discodeit.dto.request.BinaryContentCreateRequest;
+import com.sprint.mission.discodeit.dto.request.MessageCreateRequest;
+import com.sprint.mission.discodeit.dto.request.MessageUpdateRequest;
+import com.sprint.mission.discodeit.entity.Message;
+import com.sprint.mission.discodeit.service.MessageService;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
-@RestController
 @RequiredArgsConstructor
-@RequestMapping("/api/message")
-@Tag(name = "MessageController", description = "MessageController API 목록")
+@RestController
+@RequestMapping("/api/messages")
 public class MessageController {
 
-  private final BasicMessageService messageService;
-
-  @GetMapping("/{messageId}")
-  public ResponseEntity<Void> showInfo(@PathVariable UUID messageId) {
-    MessageDto messageDto = new MessageDto();
-    messageDto.setMessageId(messageId);
-
-    try {
-      messageService.showInfoMessage(messageDto);
-    } catch (IOException | ClassNotFoundException e) {
-      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
-    }
-
-    return ResponseEntity.status(HttpStatus.OK).build();
-
-  }
-
-  @GetMapping("/channel/{channelName}")
-  public ResponseEntity<Void> showInfoOfChannelMessage(@PathVariable String channelName) {
-    MessageDto messageDto = new MessageDto();
-    messageDto.setChannelName(channelName);
-
-    try {
-      messageService.showInfoChannelMessage(messageDto);
-    } catch (IOException | ClassNotFoundException e) {
-      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
-    }
-
-    return ResponseEntity.status(HttpStatus.OK).build();
-
+  private final MessageService messageService;
+
+  @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+  public ResponseEntity<Message> create(
+      @RequestPart("messageCreateRequest") MessageCreateRequest messageCreateRequest,
+      @RequestPart(value = "attachments", required = false) List<MultipartFile> attachments
+  ) {
+    List<BinaryContentCreateRequest> attachmentRequests = Optional.ofNullable(attachments)
+        .map(files -> files.stream()
+            .map(file -> {
+              try {
+                return new BinaryContentCreateRequest(
+                    file.getOriginalFilename(),
+                    file.getContentType(),
+                    file.getBytes()
+                );
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            })
+            .toList())
+        .orElse(new ArrayList<>());
```

Comment:
```suggestion
        .orElse(Collections.emptyList());
```
새로운 객체를 생성하기 보다는 만들어둔 빈 리스트를 사용하면 어떨까요?
큰 성능 차이는 없겠지만 attachmentRequests의 불변성을 지킬 수 있을듯 합니다!